src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/Transformers/AvaloniaXamlIlDuplicateSettersChecker.cs

```diff
@@ -0,0 +1,44 @@
+using System.Collections.Generic;
+using System.Linq;
+using XamlX;
+using XamlX.Ast;
+using XamlX.Transform;
+
+namespace Avalonia.Markup.Xaml.XamlIl.CompilerExtensions.Transformers;
+
+class AvaloniaXamlIlDuplicateSettersChecker : IXamlAstTransformer
+{
+    public IXamlAstNode Transform(AstTransformationContext context, IXamlAstNode node)
+    {
+        if (node is not XamlAstObjectNode objectNode)
+        {
+            return node;
+        }
+
+        var fullName = objectNode.Type.GetClrType().FullName;
+        if (fullName is not ("Avalonia.Styling.Style" or "Avalonia.Styling.ControlTheme"))
+        {
+            return node;
+        }
+
+        var properties = objectNode.Children
+            .OfType<XamlAstObjectNode>()
+            .Where(n => n.Type.GetClrType().Name == "Setter")
```

Comment:
Probably not necessary here, but this Setter type also could be checked via `context.GetAvaloniaTypes()`, even without IsAssignableFrom.

Reply:
As we technically support custom setters, and custom Setter from another namespace can be used there. Which might not necessarily work well with this validation.